src/Merchants/Content/Merchant/Services/RegistrationService.php

```diff
@@ -54,6 +56,10 @@ public function registerMerchant(array $parameters, SalesChannelContext $salesCh
     {
         $parameters['id'] = Uuid::randomHex();
 
+        if ($this->doesExists($parameters['email'])) {
```

Comment:
I don't like that name.
Maybe `isMailAvailable` and then invert it.

`if (!$this->isMailAvailable($parameters['email']))`

or something like that.